src/Symfony/Component/Console/Application.php

```diff
@@ -1101,6 +1113,33 @@ private function findAlternatives(string $name, iterable $collection): array
         return array_keys($alternatives);
     }
 
+    /**
+     * Removes hidden commands commands from a given array of command names and
+     * returns them as Command instances.
+     *
+     * @param string[] $commandNames The command names to filter
+     */
+    private function removeHiddenCommands(array &$commandNames): array
```

Comment:
I would prefer to return a list of commands instead of changing the array given as an argument.

Reply:
What about sth like this? (returning the filtered command names, but with an option to get the hidden command instances out)
```php
private function removeHiddenCommands(array $commandNames, &$hiddenCommands = null): array
```

Without that option it would be way more code duplication I guess. Otherwise I could probably as well inline the function or make it a callback if you prefer this.

Reply:
Or you can return an array of arrays. Inlining might be the best options.